### PR TITLE
Add contextual help for NPI definition

### PR DESCRIPTION
**Pull request by brainwane**
The lower case `i`, surrounded by a circle, is a commonly understood
icon for "information" so users can hover over it to get contextual
help. This commit adds two versions (one in black and one in blue) of
the "Info" icon, CC BY 3.0 by Weltenraser , from
https://thenounproject.com/term/info/1017664/ .

Relevant to #265.

### CODE REVIEW

**Comment by jasonaowen** [comment]
I suspect we need to add license information to `LICENSE.md` (cc @jvasile ).

@brainwane: would you mind using these in at least one place as part of this pull request? Adding unused files doesn't feel great to me.

**Comment by brainwane** [comment]
Ah, totally understood, Jason. Added licensing info; will use icon(s) in at least one template and will then remove the WIP prefix from the PR title.

**Comment by brainwane** [comment]
This PR now also uses one of these icons in a provider enrollment template, `psm-app/cms-web/WebContent/WEB-INF/pages/provider/enrollment/steps/pageTemplates/default/personal_information.jsp`. This means that I can use the `<abbr>` tag for the abbreviation "NPI" itself, and use the `title` attribute on the icon for longer contextual help.

npi-icon-used
npi-i-icon-hover
npi-abbrev-hover

**Comment by jvasile** [comment]
License info in License.md is appropriate but we might at some point also consider having it in the front end.  It's not a license requirement, but I've always thought that the attribution clause should result in something end-user visible in stuff like this.  Maybe a late-stage clean up step would take the license.md file and generate a "Thanks" page.  That's a future todo, so just doing license.md works for now.

**Comment by brainwane** [comment]
Conversation in IRC's demonstrating that we aren't happy with the placement of this icon (especially in its positioning and its proximity to the Required Field asterisk) and might also want to change its color. I'm looking at https://standards.usa.gov/components/form-templates/ for ideas to improve the look-and-feel.

**Comment by brainwane** [comment]
We might want to follow [the US government style guide](https://standards.usa.gov/components/form-templates/):

> Only label the optional inputs. Users can infer that all the others are required.

But until and unless we have that conversation, I'm sticking with the red asterisk for required fields. I've found that moving the asterisk to the right of the field itself, like in [this US federal web form](https://webforms.fec.gov/wfja/form99), declutters the form and makes the (i) icon more usable.

asterisk-on-right


I have tried fixing the vertical alignment of the (i) icon but haven't succeeded -- I've tried specifying `vertical-align:middle` in `reset.css` via a `div` or `span` class, and I've tried specifying it directly in the image invocation, but haven't been able to persuade the actual page to align the image in the middle of its row vertically. Fiddling with the image size might also be helpful. I welcome advice on this front.

**Comment by jasonaowen** [comment]
While we're bikeshedding, I really like the red asterisks on the right - that last screenshot looks good!

**Comment by cecilia-donnelly** [comment]
@brainwane I added `vertical-align: middle` to the `img` item in `reset.css` and it worked in Firefox and Chromium -- it's a little strange because `vertical-align: baseline` is given in the first set of commands in `reset.css`, for divs, spans, headers, etc.

It would be better to add a new class to the info elements, though, rather than just aligning all images in the middle.

**Comment by brainwane** [comment]
Thanks, @cecilia-donnelly. New Firefox screenshot after I took your advice:

middle-aligned-i-icon

**Comment by brainwane** [comment]
I've followed @cecilia-donnelly's suggestion to refactor this behavior into a *class* instead of applying it to all `img`s. `reset.css` already had a camelCased class (`clearFixed`) so I made an `infoIcon` class and the behavior still works:

middle-aligned-with-class-selector

So I think this is ready to merge.

File: LICENSE.md [hunk 1]
```diff
@@ -358,6 +358,265 @@ shall not be used in advertising or otherwise to promote the sale, use
...
+> written consent of Creative Commons. Any permitted use will be in
+> compliance with Creative Commons' then-current trademark usage
+> guidelines, as may be published on its website or otherwise made
+> available upon request from time to time. For the avoidance of doubt,
+> this trademark restriction does not form part of the License.
+>
+> Creative Commons may be contacted at <https://creativecommons.org/>.
+
+Work under CC BY includes:
+"Info" by Weltenraser
```
Comment (jasonaowen) [inline LICENSE.md]: Should this be a link to the Noun Project page?

**Comment by brainwane** [comment]
I've just discovered https://github.com/OpenTechStrategies/psm/issues/265#issuecomment-324767376 that there *is* JavaScript and JSP tooling for reasonably nicely formatted contextual help modal pop-ups in the PSM, so I'm inclined to revise this PR to reuse that functionality. I'll get to that tomorrow.

**Comment by brainwane** [comment]
I gave the JavaScript a go but right now when I click the question mark no modal pops up. See my WIP commit e6becb1 in case you want to try fiddling with it. I need to put this down to meet a September 1st deadline on some documentation I'm writing, but people are welcome to try this out and comment with tips, ideas, and fixes!

**Comment by brainwane** [comment]
In this very revised PR, we reuse the existing CSS and JavaScript infrastructure to provide a modal with a definition and help, discoverable through a question mark next to the abbreviation "NPI".
 
npi-modal

Thanks to @cecilia-donnelly for in-depth pair programming and leading diagnosis and implementation on this.

File: psm-app/cms-web/WebContent/js/script.js [hunk 2]
```diff
@@ -1882,9 +1882,14 @@ $(document).ready(function() {
     });
 
 
-    /*show definition modal*/
+    /*show definitions modal*/
     $('a.definition').live('click',function(){
-        addressLoadModal('#definitionsModal');
+        addressLoadModal('.definitionsModal');
```
Comment (jasonaowen) [inline psm-app/cms-web/WebContent/js/script.js]: I'm not sure I understand this change... the definitions modal defined in `definitions_modal.jsp` has both an `id` and a `class` of `definitionsModal`, but the NPI definition element also has a class of `definitionsModal`. Wouldn't we want the more-specific selector here?

Comment (jasonaowen) [reply psm-app/cms-web/WebContent/js/script.js]: Yes, it turns out we do: clicking on [the link that pops up the original definitions modal](https://github.com/OpenTechStrategies/psm/issues/265#issuecomment-324767376) results in both the old definitions and the new NPI modals to show:

screenshot 2017-09-13 10 55 29

File: psm-app/cms-web/WebContent/WEB-INF/pages/provider/enrollment/steps/pageTemplates/default/personal_information.jsp [hunk 3]
```diff
@@ -31,43 +32,45 @@
...
                 <c:set var="formName" value="_02_lastName"></c:set>
                 <c:set var="formValue" value="${requestScope[formName]}"></c:set>
                 <input type="text" class="normalInput" name="${formName}" value="${formValue}" maxlength="45"/>
+                <span class="required">*</span>
             </div>
             <div class="row requireField">
                 <%-- BUGR-9673 (optional NPI for some provider types) --%>
-                <label
-                    title="<%@ include
-                        file="/WEB-INF/pages/includes/userhelp/npi.jsp" %>">NPI
```
[the remark this answers is not in the document]
Comment (brainwane) [reply psm-app/cms-web/WebContent/WEB-INF/pages/provider/enrollment/steps/pageTemplates/default/personal_information.jsp]: Thanks, good catch. Done.

File: psm-app/cms-web/WebContent/js/script.js (hunk 2 above)
Comment (brainwane) [reply psm-app/cms-web/WebContent/js/script.js]: Thanks @jasonaowen -- fixed. @cecilia-donnelly you may want to also cast your eyes over this.

File: psm-app/cms-web/WebContent/WEB-INF/pages/includes/userhelp/npi_explanation.jsp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+<!-- /#tabSection -->
+
+<div class="clear"></div>
+
+</div>
+<!-- End .content -->
+</div>
+</div>
+<!-- /#printModal-->
+<!-- /#printModal-->
```
Comment (jasonaowen) [inline psm-app/cms-web/WebContent/WEB-INF/pages/includes/userhelp/npi_explanation.jsp]: I expect these comments were in the original definition modal used as a template, but they don't seem to add much. I think maybe more useful would be to indent nested content.

File: psm-app/cms-web/WebContent/WEB-INF/pages/includes/userhelp/npi_explanation.jsp [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    JSP Fragment for common NPI explanation modal.
+
+ --%>
+<!-- /#saveAsDraftModal-->
+<div id="NPIdefinitionModal" class="outLay NPIdefinition definitionsModal">
+<div class="inner">
+<!-- title -->
+<div class="modal-title">
+    <a href="javascript:;" class="greyBtn"><span class="btR"><span class="btM"><img src="<c:url value="/i/icon-x.png" />"
+                                                                                    alt=""/>Close</span></span></a>
```
Comment (jasonaowen) [inline psm-app/cms-web/WebContent/WEB-INF/pages/includes/userhelp/npi_explanation.jsp]: These two lines are pretty long. Maybe put the `img` element and the "Close" text on their own line?

File: psm-app/cms-web/WebContent/WEB-INF/pages/provider/enrollment/steps/pageTemplates/default/personal_information.jsp [hunk 6]
```diff
@@ -42,17 +42,16 @@
...
-                        <span class="required">${requireNPI ? '*' : ''}
-                        </span>
-                </label>
+                <label><abbr title="National Provider Identifier">NPI</abbr>
+                       <span><a href="javascript:" class="NPIdefinition">?</a></span>
+                    </label>
```
Comment (jasonaowen) [inline psm-app/cms-web/WebContent/WEB-INF/pages/provider/enrollment/steps/pageTemplates/default/personal_information.jsp]: Please indent the close tag to the same level as the open tag.

File: psm-app/cms-web/WebContent/WEB-INF/pages/includes/userhelp/npi_explanation.jsp (hunk 5 above)
Comment (brainwane) [reply psm-app/cms-web/WebContent/WEB-INF/pages/includes/userhelp/npi_explanation.jsp]: Good point. Fixed.

File: psm-app/cms-web/WebContent/WEB-INF/pages/includes/userhelp/npi_explanation.jsp (hunk 4 above)
Comment (brainwane) [reply psm-app/cms-web/WebContent/WEB-INF/pages/includes/userhelp/npi_explanation.jsp]: Thanks for the tip. I had basically copied and pasted this file from a different modal; now I've actually removed extraneous tags and properly indented the nested content.

File: psm-app/cms-web/WebContent/WEB-INF/pages/provider/enrollment/steps/pageTemplates/default/personal_information.jsp (hunk 6 above)
Comment (brainwane) [reply psm-app/cms-web/WebContent/WEB-INF/pages/provider/enrollment/steps/pageTemplates/default/personal_information.jsp]: Thanks! Fixed.

**Pull request merged**